test/socket.js

```diff
@@ -44,4 +44,19 @@ describe('socket', function(){
     sock.close();
   });
 
+  it('should throw a javascript error if it hits the system file descriptor limit', function() {
+    var i, socks = [], numSocks = 10000;
+    function hitlimit() {
+      for (i = 0; i < numSocks; i++) {
+        socks.push(zmq.socket('router'));
+      }
+    }
+    hitlimit.should['throw'];
+    for (i = 0; i < socks.length; i++) {
+      if (socks[i]) {
```

Comment:
I would even remove the if-statement. Every entry *has* to be a socket. If not, something went wrong (like zmq.socket returning `null` or something) and I would like the test to fail.